### ISSUE TITLE
[Community Feedback Needed] Uniform top-level file structure?

### ISSUE THREAD

**Issue by AndreMaz** [Microsoft/WebTemplateStudio#1189]
Original Issues: https://github.com/microsoft/WebTemplateStudio/issues/1124#issuecomment-531571687 https://github.com/microsoft/WebTemplateStudio/issues/1124#issuecomment-535902279

At this moment ([Release 0.0.1925201](https://github.com/microsoft/WebTemplateStudio/releases/tag/0.0.1925201) + everything at [dev branch](https://github.com/microsoft/WebTemplateStudio/compare/0.0.1925201...9fe5dca)), the file structure of generated projects is not consistent.  To show you what I mean check the top-level file structure for `Angular+Node`, `React+Node`, `Vue+Node` and `Vue+Moleculer`
![image](https://user-images.githubusercontent.com/9802754/64922533-28d52880-d7c8-11e9-9e02-cfe7007cb0f6.png)

**Current Issues**: 
- `e2e` folder is only present in Angular+Node. But where would the unit tests go? Into a new `tests` folder? In this case we would have 2 top-level folders (`tests` and `e2e`) with tests. 

- In Node and Flask projects the app logic is located in `server` folder while in case of Moleculer it resides in `services` folder. Moleculer is a microservices framework so it would be wrong to place services in `server` folder just to keep consistency with Node and Flask.

- `src`. Source of what? Frontend or backend? In case of Node/Flask I can infer that `server` folder must contain backend logic and, therefore, `src` should contain frontend. However, in case of Vue+Moleculer (`src` and `services`) things are harder.

**Now the question**: should the generated projects have a uniform top-level file structure?
For example, something like this:
```
.
├── backend/            - Backend related files
│ ├── tests/            - Backend unit tests would be placed here
│ ├── data/             - Data samples
│ └── ***               - Any other file or folder used in backend
├── frontend/           - Frontend related files
│ ├── tests/            - Frontend unit tests would be placed here
│ ├── components/       - Components folder
│ └── ***               - Any other file or folder used in frontend
├── README.md           - README file        
├── tests/              - E2E tests would be placed here
└── ***                 - Project config files or folder(s) that are related to backend and frontend
```

> `backend` is a more neutral term than `server` or `services` as currently used in Node/Flask and Moleculer templates. Also, `frontend` as a folder name is more explicit than `src`. 

**Pros of uniform structure**: Any generated project would feel familiar regardless of tech stack. 

**Cons of uniform structure**: Each framework has its own way of structuring things. Uniform structure would mess their "natural" order

**Comment by AndreMaz** [comment]
I'll go first. 

I think that uniform structure is good for beginners or people less experienced in web development.
Uniform structure will cause less cognitive load while playing around with different frameworks and developing prototypes. However, experienced developers that are familiar with web tech might/will not like it because old habits die hard.

In the end, I think that it's all about the target audience of WebTS. 
If WebTS targets the beginners (which I think it does) then it should go with uniform structure.

If the target are experienced/hardcore programmers then things should stay as they are. However, I think that the majority of experienced programmers has their own template that they re-use from project to project. So I think they won't be much interested in WebTS.

**Comment by sibille** [comment]
Uniform top level file structure was implemented as part of  #1494.
